### PR TITLE
Unify relative range preset dropdown usage in search bar and date time range picker.

### PR DESCRIPTION
**Pull request by linuspahl**
## Motivation and Context

We recently implemented a relative range preset select for the search bar (https://github.com/Graylog2/graylog2-server/pull/10740.) In a follow up PR (https://github.com/Graylog2/graylog2-server/pull/10769) we tried to unify the search time range update of the relative range preset dropdown with the date time picker.

This PR is cleaning up both changes, we are now:
- also using the `RangePresetSelect` component for the date time picker. The component is already written in a way it can be reused, but we forgot to implement it for the date time picker.
- fixing the format of the time range update when selecting a range preset. We are now:
Using the following format when selecting the preset "all messages": `{ type: "relative", range: 0 }` and in every other case: `{ type: "relative", from: 300 }`. This way the time range update of the relative range preset dropdown is unified with the update triggered by the date time range picker. Before this change the relative preset dropdown always used the following format: `{ type: "relative", range: 300 }`.
I realized this change is necessary, after finding following bug: When selecting a relative range preset option like "30 minutes" twice, the search bar form was dirty and it was not possible to get rid of the dirty state by submitting the form.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Refactoring (non-breaking change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.

### CODE REVIEW

File: graylog2-web-interface/src/views/components/TimerangeForForm.ts [hunk 1]
```diff
@@ -22,8 +22,21 @@ import type { TimeRange } from 'views/logic/queries/Query';
 import type { SearchBarFormValues } from 'views/Constants';
 import { isTypeRelativeWithStartOnly, isTypeRelativeWithEnd } from 'views/typeGuards/timeRange';
 
+import { TimeRangeDropDownFormValues } from './searchbar/date-time-picker/TimeRangeDropdown';
+
 const formatDatetime = (datetime) => datetime.toString(DateTime.Formats.TIMESTAMP);
 
+export const onSettingSearchBarFormTimeRange = (nextTimeRange: TimeRangeDropDownFormValues['nextTimeRange']) => {
```
Comment (dennisoelkers) [inline graylog2-web-interface/src/views/components/TimerangeForForm.ts:29]: The naming implies that this is just an event handler, but I think what it is actually doing is normalizing a time range by converting the "all messages" time range. If my impression is correct, I would suggest moving this to a separate module and rename it to something like `normalizeIfAllMessagesRange`.

File: graylog2-web-interface/src/views/components/TimerangeForForm.ts [hunk 2]
```diff
@@ -22,8 +22,21 @@ import type { TimeRange } from 'views/logic/queries/Query';
...
+  if (isTypeRelativeWithEnd(nextTimeRange) && nextTimeRange.from === RELATIVE_ALL_TIME) {
```
Comment (dennisoelkers) [inline graylog2-web-interface/src/views/components/TimerangeForForm.ts:30]: I think we can also create a `isAllMessagesRange` predicate. This way we can also encapsulate the usage of the constant.

**Review by dennisoelkers — CHANGES_REQUESTED** [review]
Looks good, just some code style suggestions.

File: graylog2-web-interface/src/views/components/TimerangeForForm.ts (hunk 2 above)
Comment (linuspahl) [reply graylog2-web-interface/src/views/components/TimerangeForForm.ts]: Good idea, I created a helper function, but did not started to implement it everywhere, to keep this PR small.
I am not completely sure about the best place for this type of functions and created a separate file in `views/logic/queries`.

**Review by dennisoelkers — APPROVED** [review]
Thanks! Looks good and works great now! :tada:

As a future improvement, I would suggest bundling up the different functions used for timeranges (`normalizeIfAllTimesRange`, `isAllTimesRange`, ...) in a `timerange` module.

**Pull request merged**